lib/smart_answer/calculators/part_year_profit_calculator.rb

```diff
@@ -0,0 +1,29 @@
+module SmartAnswer
+  module Calculators
+    class PartYearProfitCalculator
+      attr_accessor :tax_credits_awarded_on, :accounts_start_on, :profit_for_current_period
+
+      def initialize(**args)
+        @tax_credits_awarded_on = args[:tax_credits_awarded_on]
+        @accounts_start_on = args[:accounts_start_on]
+        @profit_for_current_period = args[:profit_for_current_period]
+      end
+
+      def basis_period
+        DateRange.new(begins_on: accounts_start_on, duration: 1.year)
+      end
+
+      def award_period
+        DateRange.new(begins_on: TaxYear.current.begins_on, ends_on: tax_credits_awarded_on)
+      end
+
+      def profit_per_day
+        (profit_for_current_period / basis_period.number_of_days).floor(2)
```

Comment:
If we were to do that, it feels as if we should also do it for the other (`Date`) attributes. Also we should do it for the `attr_writer` methods. I prefer to leave this as is it for now and rely on (a) the calculator unit test to pass in the correct types of objects; and (b) the flow unit test to verify that passing in a string to the `money_question` results in an instance of `Money` being set on the calculator. I hope that makes sense.